docs/spec.rst

```diff
@@ -0,0 +1,501 @@
+JSpecify 0.1 spec outline
+=========================
+
+About this document
+-------------------
+
+This document summarizes our current thinking for proposing a set of
+nullness annotations. **Nothing here is “set in stone” yet.**
+
+It will be very helpful to read or refer to the `core
+concepts/glossary <https://docs.google.com/document/d/1KQrBxwaVIPIac_6SCf--w-vZBeHkTvtaqPSU_icIccc/edit>`__
```

Comment:
any chance we could put the glossary into .md form as well?  The docs aren't public, but this page will be publicly visible, IIUC, so links to the docs are probably ill-advised in any case

Reply:
That is something we'll need to do. I am hoping to call it out of scope for 0.1: "Fortunately," it still seems unlikely that anyone outside the group will view the spec. Do you think we can get away with that? We can always release an 0.1.1 if we want to update the links to point to somewhere publicly visible for historical interest.

Reply:
SGTM.  maybe put something like "TODO: make these available with spec" into this text then?

Reply:
Done.